crm_lead_tag_sync/models/crm_lead.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    @api.constrains('tag_ids')
```

Comment:
You cannot use constrains to do that...

If you want to use something more readable than overriding write(), use inverse property on tag_ids field.

Reply:
I get that it's not really a constraint and it's conceptually wrong, but it gets the job done. I've just changed it anyways.
Out of curiosity, is there a technical reason not to do it?

Reply:
You are not aware of what implementation Odoo's ORM behind constrains(). The purpose is not to write on records. Odoo can change it (maybe invalidate cache or ....) and your code will not work anymore.